packages/block-editor/src/components/inserter/block-patterns-filter.js

```diff
@@ -49,17 +35,33 @@ export const SYNC_TYPES = {
 const getShouldDisableSyncFilter = ( sourceFilter ) =>
 	sourceFilter !== PATTERN_TYPES.all && sourceFilter !== PATTERN_TYPES.user;
 
+const getShouldDisableNonUserSources = ( category ) => {
+	return category.name === myPatternsCategory.name;
+};
+
 export function BlockPatternsSyncFilter( {
 	setPatternSyncFilter,
 	setPatternSourceFilter,
 	patternSyncFilter,
 	patternSourceFilter,
 	scrollContainerRef,
+	category,
 } ) {
```

Comment:
My thinking here is that we don't want to explicitly update the parent components `patternSourceFilter` state if the user switches to `My patterns` has otherwise it is then set to this filter when they switch to another category, so deriving it from the props here avoids them being confused if they switch from My patterns to another category without themselves having explicitly chose the user filter.

Reply:
have added a comment to source to explain this.